bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandler.java

```diff
@@ -174,4 +175,24 @@
      */
     void handleRemoval();
 
+    /**
+     * This method should return a unique identifier of the device handled by this {@link ThingHandler}.
```

Comment:
It would have been nice to be able to do something like
```
        <properties>
        	<property name="vendor">Philips</property>
        	<property name="serialNumber" representation="true"></property>
        </properties>
```
but as we have a `Map<String,String>` in the Java API, there is no way to add the info to the properties themselves.
So yes, we would need to do it as you suggested (maybe more clearly call it `representationProperty`).
The two problems that you list are imho no "real" problems - it is a convention that must be followed and we actually have the same issue for channel ids and other information that needs to be kept in sync between XML and Java.